tests/core/test_otel_logger.py

```diff
@@ -358,108 +358,134 @@ def test_timer_start_and_stop_manually_send_true(self, mock_time, metrics_consis
         )
 
     def test_gauge_set(self):
-        # Arrange
-        logger = self.stats
         metric_name = "test_metric"
         value = 42
         tags = {"env": "prod"}
 
-        # Act
-        logger.gauge(metric_name, value, tags=tags)
+        self.stats.gauge(metric_name, value, tags=tags)
 
-        # Assert
-        current_value = logger.metrics_map.poke_gauge(full_name(prefix=logger.prefix, name=metric_name), tags)
+        current_value = self.stats.metrics_map.poke_gauge(
+            full_name(prefix=self.stats.prefix, name=metric_name), tags
+        )
         assert current_value == value
 
     def test_gauge_increment(self):
-        # Arrange
-        logger = self.stats
         metric_name = "test_metric"
         count = 10
         tags = {"env": "prod"}
 
-        # Act
-        logger.gauge(metric_name, count, delta=True, tags=tags)  # Use delta=True to increment the gauge
+        self.stats.gauge(metric_name, count, delta=True, tags=tags)  # Use delta=True to increment the gauge
 
-        # Assert
         # Retrieve the current value using poke_gauge
-        current_value = logger.metrics_map.poke_gauge(full_name(prefix=logger.prefix, name=metric_name), tags)
+        current_value = self.stats.metrics_map.poke_gauge(
+            full_name(prefix=self.stats.prefix, name=metric_name), tags
+        )
         assert current_value == count
 
     def test_gauge_decrement(self):
-        # Arrange
-        logger = self.stats
         metric_name = "test_metric"
         count = 5
         tags = {"env": "prod"}
 
-        # Act: Increment the gauge by 'count'
-        logger.gauge(metric_name, count, delta=True, tags=tags)
+        self.stats.gauge(metric_name, count, delta=True, tags=tags)
 
-        # Assert: Check if the current value after incrementing is equal to 'count'
-        current_value_after_increment = logger.metrics_map.poke_gauge(
-            full_name(prefix=logger.prefix, name=metric_name), tags
+        current_value_after_increment = self.stats.metrics_map.poke_gauge(
+            full_name(prefix=self.stats.prefix, name=metric_name), tags
         )
         assert current_value_after_increment == count  # Ensure the gauge value reflects the increment
 
-        # Act for decrement
-        logger.gauge(metric_name, -count, delta=True, tags=tags)  # Decrement the gauge by 'count'
+        self.stats.gauge(metric_name, -count, delta=True, tags=tags)  # Decrement the gauge by 'count'
 
-        # Assert: Ensure the gauge value reflects the decrement to zero
-        current_value_after_decrement = logger.metrics_map.poke_gauge(
-            full_name(prefix=logger.prefix, name=metric_name), tags
+        current_value_after_decrement = self.stats.metrics_map.poke_gauge(
+            full_name(prefix=self.stats.prefix, name=metric_name), tags
         )
         assert current_value_after_decrement == 0
+
```

Comment:
Non-blocking nitpick/style suggestion, feel free not to implement this idea;

You could combine all of the following new tests into one parametrized test which passes in the name being tested, the tags (or perhaps a bool indicating whether to use a set of tags which are defined in the test itself), and the expected exception message.   If you'd like to see what that might look like, check out [this](https://github.com/apache/airflow/blob/main/providers/tests/amazon/aws/operators/test_appflow.py#L212) test